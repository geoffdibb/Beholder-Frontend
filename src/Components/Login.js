import React from 'react';
import { Button, InputGroup, FormText, Form, FormGroup, Label, InputGroupText, InputGroupAddon, Input } from 'reactstrap';

export default class Login extends React.Component {

    render() {
        return (
            <div>
                <InputGroup>
                    <InputGroupAddon addonType="prepend">
                        <InputGroupText>Username</InputGroupText>
                    </InputGroupAddon>
                    <Input />
                </InputGroup>

                <InputGroup>
                    <InputGroupAddon addonType="prepend">
                        <InputGroupText>Password</InputGroupText>
                    </InputGroupAddon>
                    <Input />
                </InputGroup>
                <br />
                <FormGroup check>
                    <Label check>
                        <Input type="checkbox" />
                        Verify humanity?
                    </Label>
                </FormGroup>

                <br />
                <Button color="primary">Login</Button>
            </div>
        );
    };

}