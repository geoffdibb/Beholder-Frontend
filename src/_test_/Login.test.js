import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Login from '../Components/Login';

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';


Enzyme.configure({ adapter: new Adapter() })

it('snapshot login', () => {
    const tree = renderer.create(<Login/>).toJSON()
    expect(tree).toMatchSnapshot();
  })