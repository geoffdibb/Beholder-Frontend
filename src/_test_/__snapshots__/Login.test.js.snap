// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot login 1`] = `
<div>
  <img
    alt="Beholder Logo"
    className="Logo"
    src="logo.png"
  />
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4 offset-sm-4"
      >
        <h1>
          Welcome
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6 offset-sm-3"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
              style={
                Object {
                  "width": "100px",
                }
              }
            >
              Username
            </span>
          </div>
          <input
            className="form-control"
            id="username"
            name="username"
            placeholder="username"
            type="username"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6 offset-sm-3"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          />
          <span
            className="input-group-text"
            name="password"
            style={
              Object {
                "width": "100px",
              }
            }
            type="password"
          >
            Password
          </span>
          <input
            className="form-control"
            id="Password"
            name="password"
            placeholder="password"
            type="password"
          />
          <div
            className="invalid-feedback"
          />
        </div>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-sm-4 offset-sm-4"
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <input
          checked={false}
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        Check here if you are not a bot?
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-sm-4 offset-sm-4"
      >
        <button
          aria-label={null}
          className="btn btn-dark disabled"
          disabled={true}
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </div>
  </form>
</div>
`;
